src/cli/run.ts

```diff
@@ -118,43 +120,46 @@ export const runTask = async (
   sys?: d.CompilerSystem
 ) => {
   config.flags = config.flags || { task };
+  const strictConfig: InternalStrictConfig = {...config, flags: {...config.flags} ?? { task }};
```

Comment:
`runTask` is _also_ a publicly exposed API, so I couldn't change the function signature without introducing a breaking change.   This one I'm a little more troubled by the implications of.  Specifically, we normally have validated the config by now in `run()`, which would have validated the config. However, if someone has called this directly, we'd need to find a way to 'revalidate' the config